client/src/components/CreatePost.test.tsx

```diff
@@ -55,4 +57,9 @@ describe("verifying launch of create post component", () => {
     screen.getByTestId("previewButton").click();
     expect(screen.getByTestId("PreviewPopUpComponent")).toBeInTheDocument();
   });
+
+  it("closes the dialog when clicked on `Back` button", () => {
+    screen.getByTestId("backButton").click();
+    expect(screen.getByTestId("newPostButton")).toBeInTheDocument();
```

Comment:
I think with a dialog, since it sits on top of something, the elements in the background are still in the document. Verity this by commenting out line 62 and verifying that the test fails (you don't need a specific test for this, just check whether my assumption is correct).

Reply:
Your assumption is correct! Tried it and it still worked.